geoutils/georaster/raster.py

```diff
@@ -958,12 +1002,101 @@ def copy(self: RasterType, new_array: np.ndarray | None = None) -> RasterType:
 
         return cp
 
-    @property
-    def __array_interface__(self) -> dict[str, Any]:
-        if not self.is_loaded:
-            self.load()
+    def __array__(self) -> np.ndarray:
+        """Method to cast np.array() or np.asarray() function directly on Raster classes."""
+
+        return self._data
+
+    def __array_ufunc__(
+        self,
+        ufunc: Callable[[np.ndarray | tuple[np.ndarray, np.ndarray]], np.ndarray | tuple[np.ndarray, np.ndarray]],
+        method: str,
+        *inputs: Raster | tuple[Raster, Raster],
+        **kwargs: Any,
+    ) -> Raster | tuple[Raster, Raster]:
+        """
+        Method to cast NumPy universal functions directly on Raster classes, by passing to the masked array.
+        This function basically applies the ufunc (with its method and kwargs) to .data, and rebuilds the Raster from
+        self.__class__. The cases separate the number of input nin and output nout, to properly feed .data and return
+        Raster objects.
+        See more details in NumPy doc, e.g., https://numpy.org/doc/stable/user/basics.dispatch.html#basics-dispatch.
+        """
+
+        # If the universal function takes only one input
+        if ufunc.nin == 1:
+            # If the universal function has only one output
+            if ufunc.nout == 1:
+                return self.__class__(
```

Comment:
Should we not use `self.from_array` rather than the indirect method with class?

Reply:
I don't think so because we don't have the arguments that a user can pass to `from_array` here (like a custom transform, an integer CRS).

So the only difference is the check for the transform at the beginning of `from_array`, which we don't need because the Raster object already exists and we simply copy the transform.

Reply:
What do you mean we don't have the arguments for `from_array`? You do pass a CRS and a transform, so I don't see the issue...
My only concern is that calling the Raster class with the dictionary is a trick to get `from_array` to work. There is no 100% guarantee that it will not change in the future, while the function `from_array` should always exist.  So it's safer if we use `from_array` everywhere, because then all we need to update is the `from_array` function.

Reply:
In brief, I think the call to `__class__` should never be used outside of the definition of `from_array`.

Reply:
Agree, changing to `from_array`.